src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/amendment/AmendmentToEdifactMapper.java

```diff
@@ -0,0 +1,61 @@
+package uk.nhs.digital.nhsconnect.nhais.translator.amendment;
+
+import org.apache.commons.lang3.StringUtils;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.FhirValidationException;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Segment;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentBody;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatch;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchOperation;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.JsonPatches;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public abstract class AmendmentToEdifactMapper {
+
+    private static final String REMOVE_INDICATOR = "%";
+
+    protected static boolean amendmentPatchRequiringValue(AmendmentPatch amendmentPatch) {
+        return amendmentPatch.getOp() == AmendmentPatchOperation.ADD
+            || amendmentPatch.getOp() == AmendmentPatchOperation.REPLACE;
+    }
+
+    public Optional<Segment> map(AmendmentBody amendmentBody) throws FhirValidationException {
```

Comment:
I think `throws FhirValidationException` shouldn't be here - it can always be added in subclass. Also you aren't validating FHIR.

Reply:
Inside of this method you work only on `JsonPatches` class wouldn't be easier to use `JsonPatches` as parameter for this method

Reply:
I've changed that so it accepts whole `AmendmentBody` as some mappers require NHS